src/main/scala/mimir/models/SeriesMissingValueModel.scala

```diff
@@ -0,0 +1,264 @@
+import org.joda.time.{DateTime, Seconds, Days, Period}
+import com.typesafe.scalalogging.slf4j.Logger
+
+import mimir.algebra._
+import mimir.ctables._
+import mimir.util.{RandUtils,TextUtils,TimeUtils}
+import mimir.{Analysis, Database}
+import mimir.models._
+import mimir.util._
+import mimir.statistics.DetectSeries
+
+//Upgrade: Move the series column detection to SeriesMissingValueModel
+
+object SeriesMissingValueModel
+{
+  val logger = Logger(org.slf4j.LoggerFactory.getLogger(getClass.getName))
+  val TRAINING_LIMIT = 10000
+
+  def train(db: Database, name: String, columns: Seq[String], query:Operator): Map[String,(Model,Int,Seq[Expression])] = 
+  {
+    columns.map((column) => {
+      val modelName = s"$name:$column"
+      var hintColumn: String = null
+      val model = 
+        db.models.getOption(modelName) match {
+          case Some(model) => model
+          case None => {
+            val model = new SimpleSeriesModel(modelName, column, query)
+            model.train(db)
+            db.models.persist(model)
+            hintColumn = model.getSeriesColumn
+            logger.trace(s"$column picked $hintColumn to guess ${model.assumptionFeedback}")
+            model
+          }
+      }
+      column -> (model, 0, Seq(Var(hintColumn)))
+    }).toMap
+  }
+  
+  def isTypeCompatible(columnType: Type): Boolean = 
+  {
+    val columnBase = Type.rootType(columnType)
+    columnBase match{
+      case TInt() | TFloat() | TTimestamp() | TDate() | TInterval() => true
+      case _ => false
+    }
+  }
+
+}
+/** 
+ * A model performs estimation of missing value column based on the column that follows a series.
+ * Best Guess : Performs the best guess based on the weighted-average of upper and lower bound values.
+ * Sample : Picks a random value within the range of uppper and lower bound
+ * Train : Performs best guess on missing value fields. For each missing value, a map in created
+ * 				 [ ROWID -> (Best Guess Value, Lower Bound Value, Upper Bound Value, Recieved Feedback) ] 
+ *
+ *  */
+
+@SerialVersionUID(1000L)
+class SimpleSeriesModel(name: String, colName: String, query: Operator) 
+  extends Model(name) 
+{
+  val assumptionFeedback = scala.collection.mutable.Map[String,(PrimitiveValue, PrimitiveValue, PrimitiveValue, Boolean)]()
+
+  @transient var db: Database = null
+  
+  var colType: Type = null 
+  var seriesColType: Type = null
+  var seriesColumnName: String = null
+  val colIdx:Int = query.columnNames.indexOf(colName)
+  
+  val rangeData = new Array[PrimitiveValue](2)
+
+  def train(db: Database){
+    this.db = db
+
+    val querySchema = db.bestGuessSchema(query)
+    colType = querySchema(colIdx)._2
+    
+    val findingSeries = new DetectSeries(db, 0.1)
+    val seriesColumn = findingSeries.bestMatchSeriesColumn(colName, colType, query) match {
+      case null => throw new ModelException("Query doesn't have a series Column, So this model is Invalid")        
+      case i: String => i
+    }
```

Comment:
If this model is not applicable to the given setting, then train() should just not return anything, rather than throwing an exception.  At the very least, this means that train() should have a try/catch block to pick up the `ModelException`